tests/functional/modules/test_zos_copy_func.py

```diff
@@ -468,6 +468,56 @@ def test_copy_dir_to_existing_uss_dir_not_forced(ansible_zos_module):
         hosts.all.file(path=dest_dir, state="absent")
 
 
+@pytest.mark.uss
+@pytest.mark.parametrize("src", [
+    dict(is_remote=True),
+    dict(is_remote=False),
+])
+def test_copy_subdirs_folders_and_validate_recursive_encoding(ansible_zos_module, src):
+    hosts = ansible_zos_module
+    dest_path = "/tmp/test/"
+    text_outer_file = "Hi I am point A"
+    text_inner_file = "Hi I am point B"
+    src_path = "/tmp/level_1/"
+    outer_file = "/tmp/level_1/text_A.txt"
+    inner_src_path = "/tmp/level_1/level_2/"
+    inner_file = "/tmp/level_1/level_2/text_B.txt"
+
+    try:
+        hosts.all.file(path=inner_src_path, state="directory")
+        hosts.all.file(path=inner_file, state = "touch")
+        hosts.all.file(path=outer_file, state = "touch")
+        hosts.all.shell(cmd="echo '{0}' > '{1}'".format(text_outer_file, outer_file))
+        hosts.all.file(path=dest_path, state="directory")
+        hosts.all.shell(cmd="echo '{0}' > '{1}'".format(text_inner_file, inner_file))
+
+        copy_res = hosts.all.zos_copy(content=src_path, dest=dest_path, encoding={"from": "ISO8859-1", "to": "IBM-1047"}, remote_src=src["is_remote"])
```

Comment:
How is this working when remote_src is False? Here is the deal, when src_path is created using `hosts.all.file(path=inner_src_path, state="directory")` it is acting upon the host, so local shouldn't have a folder named "/tmp/level_1/". 

Also, I don't think `src` needs to be a dict at this step

```
@pytest.mark.parametrize("src", [
    dict(is_remote=True),
    dict(is_remote=False),
])
```

Why not use? 

```
@pytest.mark.parametrize("src", [ True, False])
```

Reply:
Please review the test case when remote_src is False, I don't get it.

Reply:
What was the issue with the original test? Why was it throwing a false positive? Is that something we need to look ? Did you recreate the false positive ?

Reply:
The issue was that I put in the cat statement the path of src not of dest and then with the change I realized some more errors of paths

Reply:
Then that is that we need to look into right? If we are throwing a false positive and you've detected the issue we should have a GH issue for it at least. Could you provide more details on that?

Reply:
There is no problem in other test cases. Is this [instruction](https://github.com/ansible-collections/ibm_zos_core/blob/8e598dfe042f42c4ec3a0099d1c977fd7d85b863/tests/functional/modules/test_zos_copy_func.py#L264) where usually just copy a file there is no problem, but in my case for folders need to change a little bit, I'm don't interested in the dest_path. I need the dest_path plus the subidr.